mlflow/utils/autologging_utils/safety.py

```diff
@@ -310,6 +310,26 @@ class definition or a function object. If it is a function object, the
     else:
         assert callable(patch_function)
 
+    original_fn = gorilla.get_original_attribute(
+        destination, function_name, bypass_descriptor_protocol=False
+    )
+    # Retrieve raw attribute while bypassing the descriptor protocol
+    raw_original_obj = gorilla.get_original_attribute(
+        destination, function_name, bypass_descriptor_protocol=True
+    )
+    if original_fn != raw_original_obj:
+        raise RuntimeError(f"Unsupport patch on {str(destination)}.{str}")
+    elif isinstance(raw_original_obj, property):
+        is_property_method = True
+
+        def original(self, *args, **kwargs):
+            bound_delegate_method = raw_original_obj.fget(self)
+            return bound_delegate_method(*args, **kwargs)
```

Comment:
Can we insert an inline comment about what's going on here? We should explain why we redefine `original` and why `fget()` is used, which may not be familiar to folks reading this code.